enumerables.rb

```diff
@@ -1,37 +1,72 @@
 module Enumerables
+
   def my_each
-    count = 0
-    while count < self.size do
-      yield(self[count])
-      count += 1
+    return to_enum unless block_given?
+
+    for i in 0..self.length
+      yield(self[i])
     end
-    self
   end
 
   def my_each_with_index
-    
+    return to_enum unless block_given?
+    my_index = 0
+    self.my_each do |i|
+      yield i, my_index
+      my_index += 1
+    end
   end
 
   def my_select
     my_arr = []
-    count = 0
-    while count < self .length
-      my_arr << (self[count]) if yield(self[count]) == true
-      count += 1
-    end
+    self.my_each{ |x| my_arr << x if yield(x) }
     my_arr
   end
 
-  def my_all?
-  end
+    def my_all?
+      self.my_each do |index|
+        return true unless block_given?
+        true_false = yield(index)
+        return false unless true_false
+      end
+      true
+      to_enum
+    end
 
-  def my_any?
+  def my_any?(elem = 0)
+    unless block_given?
+      if elem.instance_of?(Class)
+        my_each { |x| return true if x.is_a? elem }
+      elsif elem.instance_of?(Regexp)
+        my_each { |x| return true if elem.match?(x.to_s) }
+      elsif [nil, false].include?(elem)
+        my_each { |x| return true if x == elem }
+      end
+      return false
+    end
+    my_each { |x| return true if yield(x) }
+    false
   end
 
   def my_none?
+    my_each do |y|
+      return false if block_given? && yield(y) || !block_given? && y
+    end
+    true
+    to_enum
   end
 
-  def my_count
+  def my_count(*arg)
+    result = 0
+    unless block_given?
+      if include?(arg)
+        my_each { |x| result += 1 if x == arg }
+        return result
+      end
+      return arg.length
+    end
+    my_each { |x| result += 1 if yield(x) }
+    result
   end
```

Comment:
- [ ]  enumerables #my_count returns the number of items in enum through enumeration
      Failure/Error: expect(array.my_count).to eq array.count
      
        expected: 100
             got: 0

- [ ]  enumerables #my_count returns the number of items in enum through enumeration::range
      Failure/Error: expect(range.my_count).to eq range.count
      
        expected: 46
             got: 0

- [ ]  enumerables #my_count counts the number of items in enum that are equal to item if an argument is given
      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
      
        expected: 10
             got: 1

- [ ] enumerables #my_count counts the number of elements yielding a true value if a block is given
      Failure/Error: let(:block) { proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 } }

- [ ] enumerables #my_count does not mutate the original array
      Failure/Error: array.my_count { |num| num + 1 }